### ISSUE TITLE
Assertion hit instead of error message in writeBlockVectorData

### ISSUE THREAD

**Issue by IshaanDesai** [precice/precice#929]
**Describe your setup**

Operating system: _Ubuntu 18.04_
preCICE Version: 2.1.1 (built in DEBUG mode)

preCICE commit: https://github.com/precice/precice/commit/fc9e02ce92d88132d36681cd40b98f46c2d86cf9

**Describe the problem**
The following assertion is hit:
```
ASSERTION FAILED
  Location:          void precice::impl::SolverInterfaceImpl::writeBlockVectorData(int, int, const int*, const double*)
  File:              /home/ishaan/precice/src/precice/impl/SolverInterfaceImpl.cpp:1078
  Rank:              0
  Failed expression: offset + dim < valuesInternal.size()
  Argument 0: 126
  Argument 1: 126
```
It is difficult to understand which argument to `writeBlockVectorData` is problematic and what is the problem.

**Step To Reproduce**
* Running the perpendicular flap tutorial case with SU2 and CalculiX here: https://github.com/precice/tutorials/tree/formatSU2-CCX-tutorial with commit: https://github.com/precice/tutorials/commit/43721e2d9f32c0b6da27c72664385f715b5f154e
* CalculiX-Adapter commit: https://github.com/precice/calculix-adapter/commit/914643c18e6249b59de4946f8a9d8631c34e64ea
* SU2-Adapter commit: https://github.com/precice/su2-adapter/commit/9761fd8e22d785592e912ccb6aad5d22e0dcaf29

**Expected behaviour**
An error message which explains what is the problem.

**Comment by fsimonis** [comment]
One of the vertex indices is incorrect.
Given the message and the config, it looks like you registered 63 vertices and tried to access vertexid `63`.
`63 * 2 + 0 = 126`, for id `63`, 2d vertex data and the component `0`.

Do you somehow process the indices before passing them back to preCICE?

This really shouldn't be an assertion, but a user error.
The interface code is a bit messy to be honest.
Furthermore, in data access methods we look for a DataContext and then totally mix up `toData` and `fromData`.
I'll attempt to clean that up.

**Comment by fsimonis** [comment]
@IshaanDesai could you try the tutorial with the version in #932? You should now run into an error instead of an assertion.

**Comment by IshaanDesai** [comment]
> Do you somehow process the indices before passing them back to preCICE?

I dont get what you mean to ask here. In the meantime I was able to find what was going wrong: I was passing passing an array of length greater than `126` for a `2D` case having `63` vertices. This was clearly wrong.

> could you try the tutorial with the version in #932?

```
---[precice] ^[[31mERROR: ^[[0m Cannot read data "Forces0" to invalid Vertex ID (272). Please make sure you only use the results from calls to setMeshVertex/Vertices().
```
The error message is clear now :+1:

**Comment by uekerman** [comment]
closed by #932
